packages/authentication/src/strategy-adapter.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import * as http from 'http';
+import {ServerRequest as Request} from 'http';
+import {HttpErrors, ParsedRequest} from '@loopback/core';
+
+/**
+ * Interface definition of a passport strategy.
+ */
+export interface Strategy {
+  authenticate(req: http.ServerRequest): void;
+}
+
+/**
+ * Shimmed Request to satisfy express requirements of passport strategies.
+ */
+export class ShimRequest {
+  headers: Object;
+  query: Object;
+  url: string;
+  path: string;
+  method: string;
+  constructor(request?: ParsedRequest) {
+    if (!request) return;
+    this.headers = request.headers;
+    this.query = request.query;
+    this.url = request.url;
+    this.path = request.path;
+    this.method = request.method;
+  }
+}
+
+/**
+ * Adapter class to invoke passport-strategy
+ *   1. provides express dependencies to the passport strategies
+ *   2. provides shimming of requests for passport authentication
+ *   3. provides lifecycle similar to express to the passport-strategy
+ *   3. provides state methods to the strategy instance
+ * see: https://github.com/jaredhanson/passport
+ */
+export class StrategyAdapter {
+  /**
+   * @param strategyCtor function which implements a passport-strategy;see: http://passportjs.org/
+   */
+  constructor(private strategyCtor: Strategy) {}
```

Comment:
The parameter name should be `strategy` as its an instance of `Strategy`, not the class/constructor.